File: HW1/javaApp/src/main/java/io/rienel/domain/interfaces/Feedable.java
package io.rienel.domain.interfaces;

import io.rienel.domain.Animal;

/**
 * todo armotozov
 *
 * @author Artem Motozov
 * @since 2022.09.19
 */
public interface Feedable<T extends Animal> {
    Boolean eat(T food);
}

File: HW1/kotlinApp/src/main/kotlin/io/rienel/domain/interfaces/Feedable.kt
package io.rienel.domain.interfaces

import io.rienel.domain.Animal

/**
 * todo armotozov
 *
 * @author Artem Motozov
 * @since 2022.09.19
 */
interface Feedable<T : Animal> {
    fun eat(food: T): Boolean
}